### PR TITLE
VS Code: Add required assets

### LINKED ISSUE #30163: Improve and fix VS Code new project instruction across getting started tutorials
In Visual Studio Code  section **Run the app** the first run says to use the combination of **control + F5** but it does not run the app, is does nothing, the right should be only **F5** (with debugger) because in the first run the Visual Studio Code will build the project .vscode directory, so the subsequents runs can use **Control + F5**

--- Added by WadePickett
Expanded this to reviewing and improving all the getting started tutorials for setup and run steps for VS Code.  (Except a separate one was made for Blazor getting started)

Move common instruction where I can into an included.

Address steps from new project, to accept authors, to accept assets to create the .vscode folder with generated launch.json and tasks.json files and verify through Ctrl + F5 step, the first run of the app.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: a25bf940-8977-fec4-461b-542a9f550fc8
* Version Independent ID: 5492c641-af15-64b8-beaa-ddf511be73e6
* Content: [Tutorial: Get started with Razor Pages in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/razor-pages-start?view=aspnetcore-7.0&tabs=visual-studio-code)
* Content Source: [aspnetcore/tutorials/razor-pages/razor-pages-start.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/razor-pages/razor-pages-start.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @wadepickett
* Microsoft Alias: **wpickett**

### PR DESCRIPTION
**Pull request by wadepickett**
Fixes #30163
.
For creating a new project particular to the tutorial, and then building and running there was an assumed step that was missing.  Added the following.

1.  Call out the add assets prompt, which occurs after the **trust the authors** dialog.
2.  Since there are a few ways it is possible to not get or miss the prompt for adding assets point out that if this step was missed or unavailable, select **View** > **Command Palette** and enter *.NET: Generate Assets for Build and Debug*. 

Moved what I could to an include that is used across the tutorials and changed the tutorial instruction around a bit to make that work in each.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/tutorials/first-mvc-app/start-mvc.md](https://github.com/dotnet/AspNetCore.Docs/blob/912233ca187a3a2f1432df28259c29fbc99437b2/aspnetcore/tutorials/first-mvc-app/start-mvc.md) | [Get started with ASP.NET Core MVC](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?branch=pr-en-us-30363) |
| [aspnetcore/tutorials/first-web-api.md](https://github.com/dotnet/AspNetCore.Docs/blob/912233ca187a3a2f1432df28259c29fbc99437b2/aspnetcore/tutorials/first-web-api.md) | [[Visual Studio](#tab/visual-studio)](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?branch=pr-en-us-30363) |
| [aspnetcore/tutorials/grpc/grpc-start.md](https://github.com/dotnet/AspNetCore.Docs/blob/912233ca187a3a2f1432df28259c29fbc99437b2/aspnetcore/tutorials/grpc/grpc-start.md) | [[Visual Studio for Mac](#tab/visual-studio-mac)](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?branch=pr-en-us-30363) |
| [aspnetcore/tutorials/razor-pages/razor-pages-start.md](https://github.com/dotnet/AspNetCore.Docs/blob/912233ca187a3a2f1432df28259c29fbc99437b2/aspnetcore/tutorials/razor-pages/razor-pages-start.md) | [[Visual Studio Code](#tab/visual-studio-code)](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/razor-pages-start?branch=pr-en-us-30363) |
| [aspnetcore/tutorials/signalr-typescript-webpack.md](https://github.com/dotnet/AspNetCore.Docs/blob/912233ca187a3a2f1432df28259c29fbc99437b2/aspnetcore/tutorials/signalr-typescript-webpack.md) | [[Visual Studio Code](#tab/visual-studio-code)](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/signalr-typescript-webpack?branch=pr-en-us-30363) |
| [aspnetcore/tutorials/signalr.md](https://github.com/dotnet/AspNetCore.Docs/blob/912233ca187a3a2f1432df28259c29fbc99437b2/aspnetcore/tutorials/signalr.md) | [[Visual Studio](#tab/visual-studio/)](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/signalr?branch=pr-en-us-30363) |

### CODE REVIEW

**Comment by guardrex** [comment]
Wade, I found that my guidance in the Blazor *Tooling* doc broke 💥 at some point due to VSC churn, so I pulled it this morning. I don't see the command palette command you mentioned coming up here with my VSC, but I did see that the following produces assets at least headed in the right direction. I don't have a test app running yet with the generated assets, but I have a tracking item to work on it when I get some time.

* For `launch.json`: **Run** > **Add Configuration** > **Edge: Launch**.
* For `tasks.json`: **Terminal** > **Configure Default Build Task** > **Create tasks.json file from template** > **.NET Core Executes .NET Core build command**.

**Comment by wadepickett** [comment]
@guardrex, thanks.  Just a side note: Installing the C# dev kit extension will conflict with the C# extension's prompt and also keep the prompt from displaying.

**Comment by guardrex** [comment]
I see. Thx. Yes, I better add that, too. I'll cross-ref your remark to my full issue.

**Comment by wadepickett** [comment]
@guardrex, I decided not to mention the C# Dev Kit conflict with the add assets prompt in this PR.  There are also a few other reasons why the prompt will not display, so rather than covering all of them I simply said how to run it manually if the prompt did not appear.

 I'm not touching the Blazor tutorials with this PR, although I did look through them to see if we were covering the steps in order or missing any.

**Comment by guardrex** [comment]
I phrased it this way for the time being ...

https://github.com/dotnet/AspNetCore.Docs/pull/30365/files

... without the C# Dev Kit mention. Even if it were mentioned, it's only a *mention*, not recommendation, given its preview status. I imagine that after release that they'll want it to be recommended.

**Comment by guardrex** [comment]
BTW ... Do you know if the command palette commands are localized into other languages. I code fenced, but if they get translated, I should covert to just bold text, as you have done here.

**Comment by guardrex** [comment]
Actually, I'm testing that now. I'll report back.

**Comment by guardrex** [comment]
😆 ... 🍀 ***Lucky Guess***&trade; 🍀 ... Here's what happened after I switched to Spanish. Yeah ... I'll leave it code fenced. I noticed this problem a lot with VS, too. When I did testing on loc in the VS UI, it was kind'a all over the place. Some stuff was localized, some not localized.

image

**Comment by wadepickett** [comment]
OK, I will go with code fenced to, thanks!

**Comment by guardrex** [comment]
Sure thing. 👍 

The menu bar ***IS*** translated, so I just went bold text for that. That part I was familiar with. VS does the same thing. After that tho, it's 🎲🎲 ***CASINO ODDS*** 🎲🎲 on translations. :smile:

**Review by tdykstra — CHANGES_REQUESTED** [review]
Looks good except for the "do you trust" question instructions.  You don't need to take my suggestion but I think we should not assume that the answer to these questions is affirmative.

File: aspnetcore/includes/vscode-trust-authors-add-assets.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Visual Studio Code displays a dialog box that asks: **Do you trust the authors of the files in this folder?**
+
+* Select:
+  * The checkbox **trust the authors of all files in the parent folder**.
+  * **Yes, I trust the authors** (because dotnet generated the files).
+
```
[the remark this answers is not in the document]
Comment (tdykstra) [reply aspnetcore/includes/vscode-trust-authors-add-assets.md:6]: And incidentally, this only just occurred to me -- I have to make the same changes to the dotnet/docs VSC tutorials.

**Review by tdykstra — APPROVED** [review]

**Pull request merged**